src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -1274,20 +1274,24 @@
     (fs/delete csr-path)
     (log/debug (i18n/trs "Removed certificate request for {0} at ''{1}''" subject csr-path))))
 
-(schema/defn revoke-existing-cert!
-  "Revoke the subject's certificate. Note this does not destroy the certificate.
-   The certificate will remain in the signed directory despite being revoked."
+(schema/defn revoke-existing-certs!
```

Comment:
Do y'all think it's okay to only have the "multiple" version of this, or is the extra overhead from the loop machinery (both here and in the Java code) problematic enough to make it worth it to use a single version for the single-revoke call?

Reply:
What extra overhead are you worried about? Are you just worried that it will now take longer to revoke a single cert? Is that something we could test?

Reply:
Yeah mostly that. I imagine it's not meaningfully slower though.

Reply:
However, it does mean I had to wrap some args in arrays unnecessarily, which could be kind of annoying clutter?

Reply:
Will that affect consumers or is it just internal clutter?

When I first read this PR, I did find it a little strange to get rid of the "single" version, but I think that's because I'm used to the multiple version just being a loop of the single version. Since this isn't that, it doesn't really seem like we need the single one?